pkgs/development/haskell-modules/configuration-ghc-8.6.x.nix

```diff
@@ -94,4 +94,12 @@ self: super: {
 
   # Only 0.6 is compatible with ghc 8.6 https://hackage.haskell.org/package/apply-refact/changelog
   apply-refact = super.apply-refact_0_6_0_0;
+
+  # ghcjs 8.6.x relies on shelly < 1.9.0 (default is 1.9.0 in current package set, adding 1.8.1).
+  shelly_1_8_1 = super.callCabal2nix "shelly" (pkgs.fetchFromGitHub {
+    owner = "gregwebs";
+    repo = "Shelly.hs";
+    rev = "d0a185c7dee96820fc3652e7603d81a28276d76c";
+    sha256 = "173msjlq6yxpvd3900ajmd8znyg9kfg5csbxbiz4m7hvxkg8kbkn";
+  }) {};
```

Comment:
Added `shelly` 1.8.1 to the GHC 8.6.x package set.

Reply:
Unfortunately we can't use `callCabal2nix` in nixpkgs because it uses [IFD](https://blog.hercules-ci.com/2019/08/30/native-support-for-import-for-derivation/).

You can see the evaluation error from ofborg because of this: https://gist.github.com/GrahamcOfBorg/cb6535e42601174683b297b6d80b9fcd

---------------------------------------

Instead, you'll have to add `shell ==1.8.1` as an `extra-package` in `pkgs/development/haskell-modules/configuration-hackage2nix.yaml`.  You'll also have to submit this PR to the `haskell-updates` branch instead of `master`.

Reply:
Thanks, @cdepillabout . Will do.